test/de/de_time_units_ago.test.ts

```diff
@@ -0,0 +1,332 @@
+    });
+
+    testSingleCase(chrono.de, "vor 5 Jahren, da sahen wir was", new Date(2012, 8 - 1, 10), (result) => {
+        expect(result.start).not.toBeNull();
+        expect(result.start.get("year")).toBe(2007);
+        expect(result.start.get("month")).toBe(8);
+        expect(result.start.get("day")).toBe(10);
+
+        expect(result.index).toBe(0);
+        expect(result.text).toBe("vor 5 Jahren");
+
+        expect(result.start).toBeDate(new Date(2007, 8 - 1, 10));
+    });
+
+    testSingleCase(chrono.de, "vor einer Woche, da sahen wir was", new Date(2012, 8 - 1, 3), (result) => {
+        expect(result.start).not.toBeNull();
+        expect(result.start.get("year")).toBe(2012);
+        expect(result.start.get("month")).toBe(7);
+        expect(result.start.get("day")).toBe(27);
+
+        expect(result.index).toBe(0);
+        expect(result.text).toBe("vor einer Woche");
+
+        expect(result.start).toBeDate(new Date(2012, 7 - 1, 27));
+    });
+
+    testSingleCase(chrono.de, "vor zwei Wochen, da sahen wir was", new Date(2012, 8 - 1, 3), (result) => {
+        expect(result.start).not.toBeNull();
+        expect(result.start.get("year")).toBe(2012);
+        expect(result.start.get("month")).toBe(7);
+        expect(result.start.get("day")).toBe(20);
+
+        expect(result.index).toBe(0);
+        expect(result.text).toBe("vor zwei Wochen");
+
+        expect(result.start).toBeDate(new Date(2012, 7 - 1, 20));
+    });
+
+    testSingleCase(chrono.de, "vorletzte Woche, da sahen wir was", new Date(2012, 8 - 1, 3), (result) => {
+        expect(result.start).not.toBeNull();
+        expect(result.start.get("year")).toBe(2012);
+        expect(result.start.get("month")).toBe(7);
+        expect(result.start.get("day")).toBe(20);
+
+        expect(result.index).toBe(0);
+        expect(result.text).toBe("vorletzte Woche");
+
+        expect(result.start).toBeDate(new Date(2012, 7 - 1, 20));
+    });
+
+    testSingleCase(chrono.de, "vorvorletzte Woche, da sahen wir was", new Date(2012, 8 - 1, 3), (result) => {
+        expect(result.start).not.toBeNull();
+        expect(result.start.get("year")).toBe(2012);
+        expect(result.start.get("month")).toBe(7);
+        expect(result.start.get("day")).toBe(13);
+
+        expect(result.index).toBe(0);
+        expect(result.text).toBe("vorvorletzte Woche");
+
+        expect(result.start).toBeDate(new Date(2012, 7 - 1, 13));
+    });
+
+    testSingleCase(chrono.de, "vor ein paar Tagen, da sahen wir was", new Date(2012, 8 - 1, 3), (result) => {
+        expect(result.start).not.toBeNull();
+        expect(result.start.get("year")).toBe(2012);
+        expect(result.start.get("month")).toBe(8);
+        expect(result.start.get("day")).toBe(1);
+
+        expect(result.index).toBe(0);
+        expect(result.text).toBe("vor ein paar Tagen");
+
+        expect(result.start).toBeDate(new Date(2012, 8 - 1, 1));
+    });
+});
+
+//test("Test - Nested time ago", function () {
```

Comment:
Same as previous comment about not leaving commented code.

If we are not ready to support the nested time ref, lets remove this and add it later.